src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -204,14 +206,66 @@ private static void SetDefaultContentRoot(WebApplicationOptions options, Configu
         }
     }
 
-    private static void ApplyDefaultAppConfiguration(WebApplicationOptions options, ConfigurationManager configuration)
+    private static void ApplyDefaultAppConfigurationSlim(IHostEnvironment env, ConfigurationManager configuration, string[]? args)
     {
+        // Logic taken from https://github.com/dotnet/runtime/blob/6149ca07d2202c2d0d518e10568c0d0dd3473576/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L229-L256
+
+        var reloadOnChange = GetReloadConfigOnChangeValue(configuration);
+
+        configuration.AddJsonFile("appsettings.json", optional: true, reloadOnChange: reloadOnChange)
+            .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: reloadOnChange);
+
+        if (env.IsDevelopment() && env.ApplicationName is { Length: > 0 })
+        {
+            try
+            {
+                var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
+                configuration.AddUserSecrets(appAssembly, optional: true, reloadOnChange: reloadOnChange);
+            }
+            catch (FileNotFoundException)
+            {
+                // The assembly cannot be found, so just skip it.
+            }
+        }
+
         configuration.AddEnvironmentVariables();
 
-        if (options.Args is { Length: > 0 } args)
+        if (args is { Length: > 0 })
         {
             configuration.AddCommandLine(args);
         }
+
+        static bool GetReloadConfigOnChangeValue(ConfigurationManager configuration)
+        {
+            const string reloadConfigOnChangeKey = "hostBuilder:reloadConfigOnChange";
+            var result = true;
+            if (configuration[reloadConfigOnChangeKey] is string reloadConfigOnChange)
+            {
+                if (!bool.TryParse(reloadConfigOnChange, out result))
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection(reloadConfigOnChangeKey).Path}' to type '{typeof(bool)}'.");
+                }
+            }
+            return result;
+        }
+    }
+
+    private static void AddDefaultServicesSlim(ConfigurationManager configuration, IServiceCollection services)
+    {
+        // Add the necessary services for the slim WebApplicationBuilder, taken from https://github.com/dotnet/runtime/blob/6149ca07d2202c2d0d518e10568c0d0dd3473576/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L266
+        services.AddLogging(logging =>
+        {
+            logging.AddConfiguration(configuration.GetSection("Logging"));
+            logging.AddSimpleConsole();
```

Comment:
I assume the decision to include the simple logger by default in the slim builder vs. the more robust console logger that's currently in `Program.cs` is intentional to reduce app size?

Reply:
It will reduce app size, eventually, once we fix SimpleConsole to not include JSON and Systemd formatters as well. I still need to log that bug in dotnet/runtime. As the code exists today in dotnet/runtime, the behavior is exactly the same between AddSimpleConsole() and AddConsole(). But I want SlimBuilder to just use the Simple format by default, and the app dev to explicitly opt-in to the other format(s) through code. My initial investigation was that this would save ~120KB

Reply:
I've opened https://github.com/dotnet/runtime/issues/85116 for this size work.